rollup.config.js

```diff
@@ -0,0 +1,47 @@
+import replace from 'rollup-plugin-replace';
+import nodeResolve from 'rollup-plugin-node-resolve';
+import commonjs from 'rollup-plugin-commonjs';
+import {uglify} from 'rollup-plugin-uglify';
+import babel from 'rollup-plugin-babel';
+
+function createConfig(file, opts = {}) {
+  const options = Object.assign({
+    production: false,
+    uglify: false
+  }, opts);
+
+  const config = {
+    input: 'index.js',
+    output: {
+      file: file,
+      format: 'umd',
+      name: 'PropTypes'
+    },
+    plugins: [
+      replace(stripEnvVariables(options.production)),
+      nodeResolve(),
+      commonjs(),
+      babel({
+        exclude: 'node_modules/**'
+      })
+    ]
+  };
+
+  if (options.uglify) {
+    config.plugins.push(uglify())
+  }
+
+  return config;
+}
+
+function stripEnvVariables(production) {
+  return {
+    __DEV__: production ? 'false' : 'true',
+    'process.env.NODE_ENV': production ? "'production'" : "'development'",
+  };
+}
+
+export default [
+  createConfig('prop-types.js'),
+  createConfig('prop-types.min.js', {uglify: true, production: true})
```

Comment:
Instead of configurable config factory it's cleaner to just write two configs and reuse `input` and `output.name` like [here](https://github.com/renatorib/react-powerplug/blob/master/rollup.config.js#L23-L62).